src/runtime/client-event-manager.js

```diff
@@ -15,33 +15,117 @@
  */
 
 import {FilterResult} from '../api/client-event-manager-api';
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import {isObject, isFunction, isEnumValue, isBoolean} from '../utils/types';
 
-/** @implements {../api/client-event-manager-api.ClientEventManagerApi}*/
+/**
+ * Helper function to describe an issue with an event object
+ * @param {!string} valueName
+ * @param {?*} value
+ * @returns {!string}
+ */
+function createEventErrorMessage(valueName, value) {
+  return 'Event has an invalid ' + valueName + '(' + value + ')';
+}
+
+/**
+ * Throws an error if the event is invalid.
+ * @param {!../api/client-event-manager-api.ClientEvent} event
+ */
+function validateEvent(event) {
+  if (!isObject(event)) {
+    throw new Error('Event must be a valid object');
+  }
+
+  if (!isEnumValue(AnalyticsEvent, event.eventType)) {
+    throw new Error(createEventErrorMessage('eventType', event.eventType));
+  }
+
+  if (!isEnumValue(EventOriginator, event.eventOriginator)) {
+    throw new Error(createEventErrorMessage('eventOriginator',
+        event.eventOriginator));
+  }
+
+  if (!isObject(event.additionalParameters)
```

Comment:
Should have caught this earlier, my bad. Would like to understand why the code below won't suffice? If it's not null, it should be an object. Otherwise, it can be null or undefined. 

if (event.additionalParameters != null &&
     !isObject(event.additionalParameters)) {
  throw error
}

Reply:
It seems like a property most workflows shouldn't really need to worry about.  I don't see any reason to make them fill in an empty {}.

Reply:
it is still unclear what your code is trying to do. if its not null and not an object throw an error. there seems to be no harm in sending {}.

Reply:
I don't see code that gives something a default value if its undefined as particularly confusing.  Seems like a fairly common operation.  I did remove it anyway because whether its there or not

Reply:
Thanks it's a lot more readable. 
nit: you might want to reverse the order. the behavior of toString on a null value is [Object null] in JavaScript and the isObject() method invokes this method internally. 

if (!!event.additionalParameters && !isObject(event.additionalParameters)
This way, if the event.additionalParameters is null, the check can exit immediately. If event.additionalParameters is not null, then actually check if it's an object.

Again don't have to do it this way. Just a nit if you agree that it looks simpler or easier to read.